tests/cephfs/xfs_test.py

```diff
@@ -0,0 +1,157 @@
+
+from tests.cephfs.cephfs_utilsV1 import FsUtils
+from utility.log import Log
+
+log = Log(__name__)
+
+"""
+Testing description:
+
+Running XFS tests on the client node and run the tests
+
+Steps to Reproduce:
+
+1. Install required packages for xfstests
+2. Mount the kernel clients
+3. Run the xfstests on the clients
+4. Verify the results of the tests
+"""
+
+
+def run(ceph_cluster, **kw):
+    try:
+        log.info("Running CephFS tests for ceph kernel xfstests")
+        # Initialize the utility class for CephFS
+        fs_util = FsUtils(ceph_cluster)
+        # Get the client nodes
+        clients = ceph_cluster.get_ceph_objects("client")
+        config = kw.get("config")
+        # Authenticate the clients
+        fs_util.auth_list(clients)
+        build = config.get("build", config.get("rhbuild"))
+        # Prepare the clients
+        fs_util.prepare_clients(clients, build)
+        client1 = clients[0]
+        fs_details = fs_util.get_fs_info(client1)
+        if not fs_details:
+            fs_util.create_fs(client1, "cephfs")
+        # check if rhel version is 8 or 9
+        rhel_version, _ = client1.exec_command(
+            sudo=True, cmd="cat /etc/os-release | grep VERSION_ID"
+        )
+        log.info("Installing epel-release depending on the RHEL version")
+        if rhel_version.startswith('VERSION_ID="8'):
+            client1.exec_command(
+                sudo=True,
+                cmd="dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm",
+            )
+        elif rhel_version.startswith('VERSION_ID="9'):
+            client1.exec_command(
+                sudo=True,
+                cmd="dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm",
+            )
+
+        commands = [
+            "yum install -y acl attr automake bc dbench dump e2fsprogs fio gawk gcc gdbm-devel"
+            " git indent kernel-devel libacl-devel libaio-devel libcap-devel libtool libuuid-devel"
+            " lvm2 make psmisc python3 quota sed sqlite udftools xfsprogs",
+            "dnf install -y glib2-devel readline-devel ncurses-devel e2fsprogs-devel gcc-c++ autoconf",
+            "git clone https://github.com/axboe/liburing.git",
+            "cd liburing && ./configure && make && make install && cd ..",
+            "yum install -y xfsdump xfsprogs-devel",
+            "yum install -y exfatprogs autoconf",
+            "git clone https://git.kernel.org/pub/scm/linux/kernel/git/jaegeuk/f2fs-tools.git",
+            "cd f2fs-tools && ./configure && make && make install && cd ..",
+            "git clone https://github.com/markfasheh/ocfs2-tools.git",
+            "cd ocfs2-tools && ./autogen.sh && ./configure && make && make install && cd ..",
+            "yum install -y libacl-devel bc libtool",
+            "git clone git://git.kernel.org/pub/scm/fs/xfs/xfstests-dev.git",
+            "cd xfstests-dev && make && sudo make install",
+        ]
```

Comment:
Could we create a Bash script with these commands? This would minimize the need to call exec_command multiple times, saving time and resources for each individual call.

Reply:
yes we can create a Bash script.  but when cleaning up, it also need a bash script at the end... so I made it exec_command on purpose..